common/src/app/common/types/container.cljc

```diff
@@ -428,12 +428,26 @@
      comps-nesting-loop?)))
 
 (defn find-valid-parent-and-frame-ids
-  "Navigate trough the ancestors until find one that is valid"
+  "Navigate trough the ancestors until find one that is valid. Returns [ parent-id frame-id ]"
   [parent-id objects children]
-  (let [parent (get objects parent-id)]
-    (if (invalid-structure-for-component? objects parent children)
-      (find-valid-parent-and-frame-ids (:parent-id parent) objects children)
-      [parent-id
-       (if (= :frame (:type parent))
-         parent-id
-         (:frame-id parent))])))
+  (letfn [(get-frame [parent-id]
+            (if (cfh/frame-shape? objects parent-id) parent-id (get-in objects [parent-id :frame-id])))
+
+          ;; Recursive function. We call it after doing some checks to the input
+          (find-valid-parent-and-frame-ids'
+            [parent-id objects children]
+            (let [parent (get objects parent-id)
+                  ;; We can always move the children to the parent they already have
+                  no-changes?
+                  (->> children (every? #(= parent-id (:parent-id %))))]
+              (if (or no-changes? (not (invalid-structure-for-component? objects parent children)))
+                [parent-id (get-frame parent-id)]
+                (recur (:parent-id parent) objects children))))]
+
+    (let [all-inside-copy? (every? #(contains? % :shape-ref) children)
+
+          parents (into #{} (keep :parent-id) children)]
+      (if (and (= 1 (count parents)) all-inside-copy?)
+        ;; If all elements are inside a copy in the same parent we just return the parent-id
+        [(first parents) (get-frame (first parents))]
+        (find-valid-parent-and-frame-ids' parent-id objects children)))))
```

Comment:
With this change, I can reorder shapes in a flex with the mouse, but not with keys up&down.

Reply:
Fixed